lib/pbench/cli/agent/commands/results/move.py

```diff
@@ -0,0 +1,196 @@
+            except BadMDLogFormat as e:
+                logger.warning("Bad Metadata.log file encountered, {}", e)
+                failures += 1
+                continue
+            except FileNotFoundError as e:
+                logger.debug("File Not Found error, {}", e)
+                failures += 1
+                continue
+            except RuntimeError as e:
+                logger.warning("Unexpected Error encountered, {}", e)
+                failures += 1
+                continue
+            except Exception as e:
+                logger.debug("Unexpected Error occurred, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt = CopyResultTb(
+                    self.context.controller, result_tb_name, self.config, logger
+                )
+            except FileNotFoundError as e:
+                logger.error("File Not Found error, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt.copy_result_tb(self.context.token)
+            except (FileUploadError, RuntimeError) as e:
+                logger.error("Error uploading a file, {}", e)
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+            except OSError:
+                logger.error("Failed to remove {}", result_tb_name)
+                failures += 1
+                continue
+
+            if copy_only:
+                copied = result_dir.parent / f"{result_dir.name}.copied"
+                try:
+                    copied.touch()
+                except OSError:
+                    logger.error("Failed to `touch` {}", copied)
+                    failures += 1
+                    continue
+            else:
+                try:
+                    shutil.rmtree(result_dir)
+                except OSError:
+                    logger.error(
+                        "Failed to remove the {} directory hierarchy", result_dir
+                    )
+                    failures += 1
+                    continue
+
+            runs_copied += 1
+
+        logger.info(
+            "Status: Total no. of tarballs {}, Successfully moved {}, Encountered {} failures",
+            no_of_tb,
+            runs_copied,
+            failures,
+        )
+
+        return 0
+
+
+@click.command()
+@common_options
+@click.option(
+    "--controller",
+    required=True,
+    envvar="_pbench_full_hostname",
+    prompt=False,
+    help="Controller name",
+)
+@click.option(
+    "--user", prompt=False, help="Pbench user (Obsolete)",
+)
+@click.option(
+    "--token",
+    required=True,
+    prompt=True,
+    envvar="PBENCH_ACCESS_TOKEN",
+    help="pbench server authentication token (will prompt if unspecified)",
+)
```

Comment:
If we're going to prompt for the token, should we also prompt for the controller?

Reply:
I am not a fan of prompting.

Existing code.